pkg/api/validation/validation.go

```diff
@@ -1088,7 +1088,7 @@ func ValidateService(service *api.Service) errs.ValidationErrorList {
 		}
 	}
 
-	for _, ip := range service.Spec.DeprecatedPublicIPs {
+	for _, ip := range service.Spec.ExternalIPs {
 		if ip == "0.0.0.0" {
 			allErrs = append(allErrs, errs.NewFieldInvalid("spec.publicIPs", ip, "is not an IP address"))
 		} else if util.IsValidIPv4(ip) && net.ParseIP(ip).IsLoopback() {
```

Comment:
Something else we should think about - can any user create an ExternalIP that is the same as the node's IP on any port?  What happens if a user tries to claim 22?  I guess that can be fixed by my other PR to hold ports open - we should try to converge those to land at roughly the same time.